client/state/reader/streams/actions.js

```diff
@@ -26,41 +26,46 @@ import {
  * @param  {object} query    The query for posts. Parameters vary by stream type.
  * @return {object}          The action object
  */
-export function requestPage( { streamKey, pageHandle } ) {
+export function requestPage( { streamKey, pageHandle, isPoll = false, gap = null } ) {
 	const indexOfColon = streamKey.indexOf( ':' );
 	const streamType = indexOfColon === -1 ? streamKey : streamKey.substring( 0, indexOfColon );
+
 	return {
 		type: READER_STREAMS_PAGE_REQUEST,
 		payload: {
 			streamKey,
 			pageHandle,
 			streamType,
+			isPoll,
+			gap,
 		},
 	};
 }
 
-export function receivePage( { streamKey, pageHandle, posts } ) {
+export function receivePage( { streamKey, pageHandle, streamItems, gap } ) {
 	return {
 		type: READER_STREAMS_PAGE_RECEIVE,
 		payload: {
 			streamKey,
-			posts,
+			streamItems,
 			pageHandle,
+			gap,
 		},
 	};
 }
 
-export function showUpdates( { streamKey } ) {
-	return {
+export const showUpdates = ( { streamKey } ) => ( dispatch, getState ) => {
```

Comment:
why does this need to be a thunk?

Reply:
ahhh, because the pending items are outside the scope of the current items reducer

Reply:
yea....its the problem with `combineReducers` IMO.  There are a lot of alternatives equally meh solutions I thought of.  One of them is for the react component to pass the action creator the `pendingItems`.  Let me know if you have any ideas